openquake/hazardlib/geo/line.py

```diff
@@ -193,6 +195,157 @@ def resample(self, section_length):
+        while 1:
+
+            # Computing distances from the reference point
+            dis = ((txy[:, 0] - rtra_prj[-1][0])**2 +
+                   (txy[:, 1] - rtra_prj[-1][1])**2 +
+                   (txy[:, 2] - rtra_prj[-1][2])**2)**0.5
+
+            # Fixing distances for points before the index
+            if idx_vtx > 0:
+                dis[0:idx_vtx] = 100000
+
+            # Index of the point on the trace with a distance just below the
+            # sampling distance
+            tmp = dis - sect_len
+            idx = np.where(tmp <= 0, dis, -np.inf).argmax()
+
+            # If the pick a point that is not the last one on the trace we
+            # compute the new sample by interpolation
+            if idx < len(dis) - 1:
+
+                tmp = [rtra_prj[-1][0], rtra_prj[-1][1], rtra_prj[-1][2]]
+                pnt = find_t(txy[idx + 1, :], txy[idx, :], tmp, sect_len)
+
+                if pnt is None:
+                    raise ValueError('Did not find the intersection')
+
+                # Update the list of coordinates
+                xg, yg = proj(np.array([pnt[0]]), np.array([pnt[1]]),
+                              reverse=True)
+                rtra.append(Point(xg, yg, pnt[2]))
+                rtra_prj.append(list(pnt))
+
+                # Updating incremental length
+                inc_len += sect_len
+
+                # Adding more points still on the same segment
+                Δx = (txy[idx + 1, 0] - rtra_prj[-1][0])
+                Δy = (txy[idx + 1, 1] - rtra_prj[-1][1])
+                Δz = (txy[idx + 1, 2] - rtra_prj[-1][2])
+                chk_dst = (Δx**2 + Δy**2 + Δz**2)**0.5
+                Δx_ratio = Δx / chk_dst
+                Δy_ratio = Δy / chk_dst
+                Δz_ratio = Δz / chk_dst
+
+                while chk_dst > sect_len * 0.9999:
+
+                    x_pnt = rtra_prj[-1][0] + Δx_ratio * sect_len
+                    y_pnt = rtra_prj[-1][1] + Δy_ratio * sect_len
+                    z_pnt = rtra_prj[-1][2] + Δz_ratio * sect_len
+
+                    # New point
+                    xg, yg = proj(np.array([x_pnt]), np.array([y_pnt]),
+                                  reverse=True)
+                    rtra.append(Point(xg[0], yg[0], z_pnt))
+                    rtra_prj.append([x_pnt, y_pnt, z_pnt])
+
+                    # Updating incremental length
+                    inc_len += sect_len
+
+                    # This is the distance between the last resampled point
+                    # and the second vertex of the segment
+                    chk_dst = ((txy[idx + 1, 0] - rtra_prj[-1][0])**2 +
+                               (txy[idx + 1, 1] - rtra_prj[-1][1])**2 +
+                               (txy[idx + 1, 2] - rtra_prj[-1][2])**2)**0.5
+
+            else:
+
+                # Adding one point
+                if tot_len - inc_len > 0.5 * sect_len and not orig_extremes:
+
+                    # Adding more points still on the same segment
```

Comment:
I don't like non-ASCII identifiers since they are difficult to enter with the keyboard (and Python is not Julia).

Reply:
PS: I fixed it myself